consensus/src/simplex/actors/voter/actor.rs

```diff
@@ -314,7 +355,42 @@ impl<
         }
     }
 
-    /// Persists our nullify vote to the journal for crash recovery.
+    /// Returns a request to certify a proposal if it should be requested.
+    ///
+    /// Returns `None` if the proposal should not be certified for any reason, for example if it
+    /// already has a certification request or does not have enough information to make the request.
+    async fn try_certify(&mut self, view: View) -> Option<Request<D, P, bool>> {
+        let (context, proposal) = self.state.try_certify(view)?;
+        let receiver = self
+            .automaton
+            .certify(context.clone(), proposal.payload)
+            .await;
+        Some(Request(context, receiver))
+    }
+
+    /// Handles the certification of a proposal.
+    ///
+    /// The certification may succeed, in which case the proposal can be used in future views—
+    /// or fail, in which case we should nullify the view as fast as possible.
+    async fn certified(&mut self, view: View, success: bool) {
+        let Some(notarization) = self.state.certified(view, success) else {
+            return;
+        };
+
+        // Persist certification result for recovery
+        let artifact = Artifact::Certification(Rnd::new(self.state.epoch(), view), success);
+        self.append_journal(view, artifact.clone()).await;
+        self.sync_journal(view).await;
+
+        // Inform listeners of successful certification
+        if success {
+            self.reporter
+                .report(Activity::Certification(notarization))
+                .await;
+        }
+    }
```

Comment:
As discussed, we should also update the `resolver` here (and remove update during normal `notarization`)